runtime/sema/gen/main.go

```diff
@@ -2069,6 +2240,100 @@ func compositeTypeLiteral(ty *typeDecl) dst.Expr {
 	}
 }
 
+func interfaceTypeExpr(ty *typeDecl) dst.Expr {
```

Comment:
Same as above: Is there anything significantly different from the composite-related functions (`compositeTypeExpr`, `compositeTypeLiteral`)? If the code generation is pretty much the same, maybe refactor it into a common function